src/ring_theory/graded_algebra/radical.lean

```diff
@@ -0,0 +1,194 @@
+
+  have not_mem_I : proj 𝒜 max₁ x * proj 𝒜 max₂ y ∉ I,
+  { have neither_mem : proj 𝒜 max₁ x ∉ I ∧ proj 𝒜 max₂ y ∉ I,
+    { rw mem_filter at mem_max₁ mem_max₂,
+      exact ⟨mem_max₁.2, mem_max₂.2⟩, },
+    intro rid,
+    cases homogeneous_mem_or_mem ⟨max₁, submodule.coe_mem _⟩ ⟨max₂, submodule.coe_mem _⟩ mem_I,
+    { apply neither_mem.1 h },
+    { apply neither_mem.2 h }, },
+
+  exact not_mem_I mem_I,
+end⟩
+
+lemma ideal.is_homogeneous.is_prime_iff {I : ideal A} (h : I.is_homogeneous 𝒜) :
+  I.is_prime ↔
+  (I ≠ ⊤) ∧
+    ∀ {x y : A}, set_like.is_homogeneous 𝒜 x → set_like.is_homogeneous 𝒜 y
+      → (x * y ∈ I → x ∈ I ∨ y ∈ I) :=
+⟨λ HI,
+  ⟨ne_of_apply_ne _ HI.ne_top, λ x y hx hy hxy, ideal.is_prime.mem_or_mem HI hxy⟩,
+  λ ⟨I_ne_top, homogeneous_mem_or_mem⟩,
+    h.is_prime_of_homogeneous_mem_or_mem I_ne_top @homogeneous_mem_or_mem⟩
+
+lemma ideal.is_prime.homogeneous_core {I : ideal A} (h : I.is_prime) :
+  (I.homogeneous_core 𝒜 : ideal A).is_prime :=
+begin
+  apply (ideal.homogeneous_core 𝒜 I).prop.is_prime_of_homogeneous_mem_or_mem,
+  { exact ne_top_of_le_ne_top h.ne_top (ideal.coe_homogeneous_core_le 𝒜 I) },
+  rintros x y hx hy hxy,
+  have H := h.mem_or_mem (ideal.coe_homogeneous_core_le 𝒜 I hxy),
+  refine H.imp _ _,
+  { exact ideal.mem_homogeneous_core_of_is_homogeneous_of_mem hx, },
+  { exact ideal.mem_homogeneous_core_of_is_homogeneous_of_mem hy, },
+end
+
+lemma ideal.is_homogeneous.radical_eq {I : ideal A} (hI : I.is_homogeneous 𝒜) :
+  I.radical = Inf { J | I ≤ J ∧ J.is_homogeneous 𝒜 ∧ J.is_prime } :=
+begin
+  letI : Π i (x : 𝒜 i), decidable (x ≠ 0) := λ i x, classical.dec _,
+  rw ideal.radical_eq_Inf,
+  apply le_antisymm,
+  { refine Inf_le_Inf _,
+    rintros J ⟨HJ₁, _, HJ₂⟩,
+    exact ⟨HJ₁, HJ₂⟩, },
+  { intros x hx,
+    rw [submodule.mem_Inf] at hx ⊢,
+    rintros J ⟨HJ₁, HJ₂⟩,
+    refine (ideal.coe_homogeneous_core_le 𝒜 J) (hx _ _),
+    refine ⟨_, subtype.prop _, HJ₂.homogeneous_core⟩,
+    refine (homogeneous_ideal.homogeneous_core_coe_eq_self ⟨I, hI⟩).symm.trans_le
+      (ideal.homogeneous_core_mono _ HJ₁), }
+end
+
+lemma homogeneous_ideal.radical_eq (I : homogeneous_ideal 𝒜) :
+  (I : ideal A).radical = Inf {J | ↑I ≤ J ∧ J.is_homogeneous 𝒜 ∧ J.is_prime} :=
+ideal.is_homogeneous.radical_eq I.2
+
+lemma ideal.is_homogeneous.radical {I : ideal A} (h : I.is_homogeneous 𝒜)  :
+  I.radical.is_homogeneous 𝒜 :=
+begin
+  have radI_eq : I.radical = _ := homogeneous_ideal.radical_eq ⟨I, h⟩,
+  rw radI_eq,
+  convert (Inf {J : homogeneous_ideal 𝒜 | I ≤ J.val ∧ J.val.is_prime}).2,
+  ext J,
+  simp only [subtype.coe_mk, set.mem_set_of_eq, subtype.exists, exists_prop],
+  split;
+  intro H,
+  { exact ⟨_, H.2.1, ⟨H.1, H.2.2⟩, rfl⟩, },
+  { obtain ⟨J', HJ1, ⟨HJ2, HJ3⟩, rfl⟩ := H,
+    exact ⟨HJ2, HJ1, HJ3⟩, },
+end
+
+/--
+Radical of any homogeneous ideal is homogeneous.
+-/
+def homogeneous_ideal.radical (I : homogeneous_ideal 𝒜) : homogeneous_ideal 𝒜 :=
+⟨I.1.radical, I.2.radical⟩
```

Comment:
This docstring is a bit off, as this is data not a proof. You should provide the obvious simp lemma too:
```suggestion
/-- The radical of a homogenous ideal, as another homogenous ideal. -/
def homogeneous_ideal.radical (I : homogeneous_ideal 𝒜) : homogeneous_ideal 𝒜 :=
⟨(I : ideal R).radical, I.prop.radical⟩

@[simp]
lemma homogeneous_ideal.coe_radical (I : homogeneous_ideal 𝒜) :
  (I.radical : ideal R) = (I : ideal R).radical := rfl
```